scrapy/downloadermiddlewares/httpcache.py

```diff
@@ -33,19 +40,19 @@ def __init__(self, settings, stats):
         self.stats = stats
 
     @classmethod
-    def from_crawler(cls, crawler):
+    def from_crawler(cls, crawler: Crawler) -> "HttpCacheMiddleware":
```

Comment:
this method doesn't return HttpCacheMiddleware in a subclass, see https://stackoverflow.com/questions/44640479/mypy-annotation-for-classmethod-returning-instance for a discussion

Reply:
Em, I can see this will be a hot topic in this typing hint topic.

First, should all kinds of possible typing hint be covered? Using `typing.Type` to cover all subclasses is not a problem. The problem is should we use `typing.Type` for all classes in Scrapy, just like the previous `Request` and `Response` typing hint in this pull request?

Second, annotating the class itself is supported until python 3.7 with the help of `__future__.annotations`. Since Scrapy is still compatible with python 3.5, what should we do for `classmethod` return at this moment? A string typing hint is not a good solution in my opinion.